sdk/core/System.ClientModel/src/Message/BinaryContent.cs

```diff
@@ -0,0 +1,179 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.ClientModel.Internal;
+using System.ClientModel.Primitives;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace System.ClientModel;
+
+public abstract class BinaryContent : IDisposable
```

Comment:
Pri 0: should this type be in the same package as BinaryData? This would allow us to inherit BinaryData from BinaryContent, which I think would be ideal.

Reply:
Interesting!  I'm not opposed to the idea if we can find a way to do it that would meet our timelines and satisfy technical requirements.  

On the latter topic - I don't think you can add static extension methods to a type, so if we wanted to have include the `BinaryContent.Create(T model) : where T is IPersistableModel<T>` method, I think we've already opted ourselves out of that by shipping those interfaces GA?

Reply:
@KrzysztofCwalina asked offline: "If we have a cast from models to BinaryData, do we still need this Create method?" -- I think that the serialization routines were planning to use this method?  But maybe not.  @m-nash, do you know offhand?

Reply:
But I think you're right, @KrzysztofCwalina, if the point of BinaryContent is to allow streaming content and that's all it adds over BinaryData, in principle, it shouldn't be needed at all for models, given their finite size.

How close does this put us to moving to BinaryData for protocol methods that don't require streaming inputs or outputs?  That would get us out of the response buffering problem we're discussing separately...

Reply:
Discussed offline: I'll send a quick note to the BCL team to see if this is something they'd entertain us adding to System.Memory.Data - if not we'll fail fast on the effort and keep it in ClientModel.

Reply:
The BCL team feels we should keep this type in ClientModel.